purchases/src/main/kotlin/com/revenuecat/purchases/Purchases.kt

```diff
@@ -1341,7 +1342,9 @@ class Purchases internal constructor(
             ProductType.SUBS,
             productIds,
             { subscriptionProducts ->
-                val productsById = subscriptionProducts.groupBy { subProduct -> subProduct.productId }.toMutableMap()
+                val productsById = subscriptionProducts
+                    .groupBy { subProduct -> subProduct.purchasingData.productId } // TODO: Maybe add parent plan to StoreProduct?
+                    .toMutableMap()
```

Comment:
This broke when `storeProduct.id` started to include the base plan id in it 😱 

The product/subscription/parent id is stored by itself in `purchasingData` so this now does the same thing as it previously did. I don't think I _love _ this though because it feels bad digging into `purchasingData` 🤔 

Thoughts?